pkg/controllers/streaming/processor_controller.go

```diff
@@ -197,6 +197,15 @@ func (r *ProcessorReconciler) reconcile(ctx context.Context, logger logr.Logger,
 	processor.Status.DeploymentName = deployment.Name
 	processor.Status.PropagateDeploymentStatus(&deployment.Status)
 
+	processor.Status.MarkStreamsReady()
+	for _, stream := range append(inputStreams, outputStreams...) {
+		if !stream.Status.IsReady() {
+			ready := stream.Status.GetCondition(stream.Status.GetReadyConditionType())
+			processor.Status.MarkStreamsNotReady(fmt.Sprintf("Stream %s is not ready: %s", stream.Name, ready.Message))
```

Comment:
In my tests, a nil dereference happens here (ready == nil)

This is with a created Stream whose provisioner address does not respond (because the provider is not there (yet))

Reply:
Can we please, please, please get those tests under CI?

Reply:
fixed